### DISCUSSION TITLE
Set specific action when clicking on a table row

### DISCUSSION THREAD

**Discussion by HenrijsS** [filamentphp/filament#13495] [Help]
### Package

Panel builder

### Package Version

v3.2.92

### How can we help you?

Hi.

The new infolist feature is great, but I have a problem showing it.
There's also the new "View" page.

So what I'm trying to achieve:
- I have a table of Users.
- When the **Row** is clicked, it opens a modal of the infolist.
- The infolist **also** has the Edit button
- The actions in each row **do not** show "View".

I found this discussion: https://github.com/filamentphp/filament/discussions/11757

The problems:
- Applying hidden to the ViewAction hides it, but then, when clicking on the record in the table, it goes straight to edit.
- Adding it back, it links to the view page, **if it exists**.
- - If it's there, but there isn't a view page, it also goes to edit page, instead of opening the modal.

It's a bit confusing for me right now as to what is the expected behaviour.

Also there's nothing mentioned in the documentation about `->recordUrl(...)` and  `->recordAction(...)`

**Comment by maxacarvalho** [comment]
Hey @HenrijsS 

I had the same issue and I solved it by:

1. Created a new Table Action to execute the logic I need
2. In the resource table config I used the method `recordAction` to set my action, for example: `->recordAction(SwitchPartnerAction::class)`
3. In the resource table `actions` array you must add the action that's going to be executed with the row click, in my case, `SwitchPartnerAction`. I also didn't want the column with the action's label to show up so I just set the label to empty `SwitchPartnerAction::make()->label('')`. It's a bit hacky but it works.
